sensors-detect: add support for FSC Teutates Chip

chris-est opened the pull request:
This patch adds support for the FTS Teutates Chip, used on Fujitsu Mainboards for Intel Skylake and later Chipsets. It is based on a patch from Fujitsu and was tested on a D-3446-S Mainboard.
The corresponding kernel module can be found [here](https://github.com/torvalds/linux/blob/master/drivers/hwmon/ftsteutates.c#L711).

This is a revised version of this [pull request](https://github.com/lm-sensors/lm-sensors/pull/9).

olysonek commented on prog/detect/sensors-detect:
```diff
@@ -1391,6 +1391,11 @@ use vars qw(@i2c_adapter_names);
+		name => "FSC Teutates",
+		driver => "ftsteutates",
+		i2c_addrs => [0x73],
+		i2c_detect => sub { fts_detect(@_, 1); },
```
Why are you passing the `1`? You're not using it in `fts_detect`.

olysonek commented on prog/detect/sensors-detect:
```diff
@@ -6364,6 +6369,33 @@ sub fsc_detect
+	# Device Detection Registers must be 0x17, 0x34, 0x54
+	return unless i2c_smbus_read_byte_data($file, 0x0C) == 0x17
+		  and i2c_smbus_read_byte_data($file, 0x0D) == 0x34
+		  and i2c_smbus_read_byte_data($file, 0x0E) == 0x54;
```
That's an odd mixture of tabs and spaces. Use tabs for indentation, spaces for alignment. I.e. one tab, the rest should be spaces.

olysonek commented on prog/detect/sensors-detect:
```diff
@@ -1391,6 +1391,11 @@ use vars qw(@i2c_adapter_names);
 		i2c_addrs => [0x73],
 		i2c_detect => sub { fsc_detect(@_, 7); },
+	}, {
+		name => "FSC Teutates",
```
I guess this should say FTS instead of FSC.

olysonek commented on prog/detect/sensors-detect:
```diff
@@ -6364,6 +6369,33 @@ sub fsc_detect
 	return 8;
 }
 
+# Chip to detect: FSC Teutates
```
FTS

chris-est commented:
Thank you for your review.
I fixed the indentation, the FSC/FTS naming and removed the parameter from fts_detect() function call.
Also updated the commit message (sorry for the multiple force-pushes).

olysonek commented:
No problem at all. Thanks!
